espree.js

```diff
@@ -3857,6 +3858,15 @@ function parseParams(firstRestricted) {
                     message = Messages.StrictParamDupe;
                 }
             }
+            if (match("=")) {
+                if (!allowDefaultParams) {
+                    throwUnexpected(lookahead);
+                }
+                lex();
+                def = parseAssignmentExpression();
+                ++defaultCount;
+            }
+            defaults.push(def);
```

Comment:
Checked and yes it is.

For example if you have `function x(a, b, c){}`

The `defaults` should equal `[]` instead of `[null, null, null]`.

The `null` are only expected as placeholders when there is a partially full list of default parameters.